sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -274,13 +286,69 @@ export class ApplicationService extends RecordDataModelService<Application> {
       application.currentAssessment.modifier = auditUser;
       application.currentAssessment.studentAssessmentStatusUpdatedOn = now;
       application.currentAssessment.updatedAt = now;
-
       await applicationRepository.save([application, newApplication]);
+      await this.saveApplicationEditedTooManyTimesNotification(
+        newApplication.applicationNumber,
+        application.studentId,
+        transactionalEntityManager,
+      );
     });
-
     return { application, createdAssessment: originalAssessment };
   }
 
+  /**
+   * Saves a notification when the application is edited too many times
+   * governed by APPLICATION_EDIT_COUNT_TO_SEND_NOTIFICATION.
+   * @param applicationNumber application number of the related application.
+   * @param studentId related student id.
+   * @param transactionalEntityManager entity manager to be a part of the transaction.
+   */
+  private async saveApplicationEditedTooManyTimesNotification(
+    applicationNumber: string,
+    studentId: number,
+    transactionalEntityManager: EntityManager,
+  ): Promise<void> {
+    const applicationRepository =
+      transactionalEntityManager.getRepository(Application);
+    const applicationsCount = await applicationRepository.count({
+      where: { applicationNumber },
+    });
+    if (applicationsCount > APPLICATION_EDIT_COUNT_TO_SEND_NOTIFICATION) {
+      const notification = await transactionalEntityManager
```

Comment:
There is a difference between requesting a count from the DB vs requesting if a record exists and we should query the DB for what we need, in this situation would be just an exists.